receiver/kafkareceiver/azureresourcelogs_unmarshaler.go

```diff
@@ -0,0 +1,205 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafkareceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/kafkareceiver"
+
+import (
+	"bytes"
+	"strconv"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/relvacode/iso8601"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	conventions "go.opentelemetry.io/collector/semconv/v1.13.0"
+	"go.uber.org/zap"
+)
+
+const (
+	// Constants for OpenTelemetry Specs
+	receiverScopeName = "otelcol/" + typeStr
+
+	// Constants for Azure Log Records
+	azureCategory          = "azure.category"
+	azureCorrelationID     = "azure.correlation.id"
+	azureDuration          = "azure.duration"
+	azureIdentity          = "azure.identity"
+	azureOperationName     = "azure.operation.name"
+	azureOperationVersion  = "azure.operation.version"
+	azureProperties        = "azure.properties"
+	azureResourceID        = "azure.resource.id"
+	azureResultType        = "azure.result.type"
+	azureResultSignature   = "azure.result.signature"
+	azureResultDescription = "azure.result.description"
+	azureTenantID          = "azure.tenant.id"
+)
+
+type azureResourceLogsUnmarshaler struct {
+	version string
+	logger  *zap.Logger
+}
+
+// azureRecords represents an array of Azure log records
+// as exported via an Azure Event Hub
+type azureRecords struct {
+	Records []azureLogRecord `json:"records"`
+}
+
+// azureLogRecord represents a single Azure log following
+// the common schema:
+// https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/resource-logs-schema
+type azureLogRecord struct {
+	Time              string       `json:"time"`
+	ResourceID        string       `json:"resourceId"`
+	TenantID          *string      `json:"tenantId"`
+	OperationName     string       `json:"operationName"`
+	OperationVersion  *string      `json:"operationVersion"`
+	Category          string       `json:"category"`
+	ResultType        *string      `json:"resultType"`
+	ResultSignature   *string      `json:"resultSignature"`
+	ResultDescription *string      `json:"resultDescription"`
+	DurationMs        *string      `json:"durationMs"`
+	CallerIPAddress   *string      `json:"callerIpAddress"`
+	CorrelationID     *string      `json:"correlationId"`
+	Identity          *interface{} `json:"identity"`
```

Comment:
Any reason to leave this as an `interface` ? would `[]byte` be better? (sorry, just trying to understand the choice made.